floris/simulation/farm.py

```diff
@@ -76,7 +81,10 @@ class Farm(BaseClass):
 
     turbine_definitions: list = field(init=False, validator=iter_validator(list, dict))
     coordinates: List[Vec3] = field(init=False)
-    turbine_fCts: tuple = field(init=False, default=[])
+
+    turbine_fCts: Dict[str, interp1d] | List[interp1d] = field(init=False, default=[])
+    turbine_fCts_sorted: NDArrayFloat = field(init=False, default=[])
```

Comment:
I've tried to create a demo of this, but I can't show it. Can you provide a concrete example?

Reply:
Sure thing, see the below use case where two classes are defined using default value, but the `default=[]` attribute shares its pointer in memory, and so both `a1.x` and `a2.x` were changed. The `factory=list` is able to avoid this because it says to create a new list every time `A` is instantiated without a user-provided value. This is highly similar, if not the same thing with default empty iterables in functions.

![image](https://github.com/NREL/floris/assets/13874373/b3728814-d689-4908-b1ee-df1f1a18738d)

Reply:
Thanks @RHammond2. This is a high priority issue, but I'll address it separately from this pull request since it will be a fairly large change. I'll open an issue and subsequent pull request. Good catch, and thanks for the additional explanation.

Reply:
You're welcome, and I had a feeling this might just be its own PR.